packages/storage-types/exp/index.d.ts

```diff
@@ -121,102 +121,112 @@ export interface ListResult {
 }
 
 /**
- * The full set of object metadata, including read-only properties.
+ * Object metadata that can be set at any time.
  * @public
  */
-export interface Metadata {
+export interface SettableMetadata {
   /**
-   * The bucket this object is contained in.
+   * Served as the 'Cache-Control' header on object download.
    */
-  bucket: string;
+  cacheControl?: string | undefined;
 
   /**
-   * The full path of this object.
+   * Served as the 'Content-Disposition' header on object download.
    */
-  fullPath: string;
+  contentDisposition?: string | undefined;
 
   /**
-   * The object's generation.
-   * {@link https://cloud.google.com/storage/docs/generations-preconditions}
+   * Served as the 'Content-Encoding' header on object download.
    */
-  generation: string;
+  contentEncoding?: string | undefined;
 
   /**
-   * The object's metageneration.
-   * {@link https://cloud.google.com/storage/docs/generations-preconditions}
+   * Served as the 'Content-Language' header on object download.
    */
-  metageneration: string;
+  contentLanguage?: string | undefined;
 
   /**
-   * The short name of this object, which is the last component of the full path.
-   * For example, if fullPath is 'full/path/image.png', name is 'image.png'.
+   * Served as the 'Content-Type' header on object download.
    */
-  name: string;
+  contentType?: string | undefined;
 
   /**
-   * The size of this object, in bytes.
+   * Additional user-defined custom metadata.
    */
-  size: number;
-
+  customMetadata?:
+    | {
+        [key: string]: string;
+      }
+    | undefined;
+}
+/**
+ * Object metadata that can be set at upload.
+ * @public
+ */
+export interface UploadMetadata extends SettableMetadata {
   /**
-   * A date string representing when this object was created.
+   * A Base64-encoded MD5 hash of the object being uploaded.
    */
-  timeCreated: string;
+  md5Hash?: string | undefined;
+}
 
+/**
+ * The full set of object metadata, including read-only properties.
+ * @public
+ */
+export interface FullMetadata extends UploadMetadata {
   /**
-   * A date string representing when this object was last updated.
+   * The bucket this object is contained in.
    */
-  updated: string;
+  bucket?: string;
```

Comment:
Since this is not used for metadata that is returned from the backed, it should no longer be optional.

Reply:
Is this guaranteed to always be returned from the backend? Is there a list of metadata properties that always come back from the backend?

Reply:
Looks like there is a list here: https://cloud.google.com/storage/docs/json_api/v1/objects#resource